### PR TITLE
feat: add boot volume encryption

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Description

Add support in SLZ for KMS boot volume encryption.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Addition of new optional key `boot_volume_crk_name` in the cluster object which will help in choosing which key can be used for boot volume encryption.
For example:
```
  "clusters": [
    {
        "cos_name": "cos",
        "entitlement": "cloud_pak",
        "kube_type": "openshift",
        "kube_version": "latest",
        "machine_type": "bx2.16x64",
        "name": "management-cluster",
        "resource_group": "slz-management-rg",
        "boot_volume_crk_name": "slz-roks-key",
        "kms_config": {
           "crk_name": "slz-roks-key",
            "private_endpoint": true
        },
        "subnet_names": [
            "vsi-zone-1",
            "vsi-zone-2",
            "vsi-zone-3"
        ],
        "update_all_workers" : false,
        "vpc_name": "management",
        "worker_pools": [
            {
                "entitlement": "cloud_pak",
                "flavor": "bx2.16x64",
                "name": "logging-worker-pool",
                "subnet_names": [
                    "vsi-zone-1",
                    "vsi-zone-2",
                    "vsi-zone-3"
                ],
                "vpc_name": "management",
                "workers_per_subnet": 2
            }
        ],
        "workers_per_subnet": 2
    }
]
```

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

**Comment by ocofaigh** [comment]
@Aashiq-J What about the override json, are we just going to use the KMS details from `clusters.kms_config` ?
AKA:
```
 "clusters": [
    {
        "cos_name": "cos",
        "entitlement": "cloud_pak",
        "kube_type": "openshift",
        "kube_version": "latest",
        "machine_type": "bx2.16x64",
        "name": "management-cluster",
        "resource_group": "slz-management-rg",
        "kms_config": {
           "crk_name": "slz-roks-key",
            "private_endpoint": true
        },
```

**Comment by ocofaigh** [comment]
Test failed with a backend issue: `Error: [0m[0m[1m[ERROR] Error Deleting resource group: Resource groups with active or pending reclamation instances can't be deleted. Use the CLI commands "ibmcloud resource service-instances --type all" and "ibmcloud resource reclamations" to check for remaining instances, then delete the instances and try again.`

Lets get https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/pull/516 merged first, and then rebase this with main and then run pipeline again

**Comment by ocofaigh** [comment]
@Aashiq-J Do we know what the impact is here on existing clusters that were provisioned in SLZ? Does it need to recreate the workers in order to encrypt them? Or can encryption be added as an in place update? Perhaps upgrade test will give us the answer

**Comment by Aashiq-J** [comment]
@ocofaigh , even though the upgrade test in the pipeline passed, will run the upgrade test manually adding more worker pools and check if anything is being destroyed and recreated.

**Comment by ocofaigh** [comment]
yes thanks @Aashiq-J - I was kind of expecting it to fail tbh

**Comment by Aashiq-J** [comment]
@ocofaigh , I have tested it multiple times and it looks like once the cluster is created we can't change the boot volume encryption. Even if we apply above existing cluster it doesn't change anything. 

And since, `boot_volume_crk_name` is a optional variable, we are not directly using the kms_config for boot_vol encryption.

File: patterns/roks/module/config.tf [hunk 1]
```diff
@@ -79,14 +79,15 @@ locals {
...
+        kube_type            = "openshift"
+        kube_version         = var.kube_version
+        resource_group       = "${var.prefix}-${network}-rg"
+        update_all_workers   = var.update_all_workers
+        cos_name             = "cos"
+        entitlement          = var.entitlement
+        boot_volume_crk_name = "${var.prefix}-roks-key"
         # By default, create dedicated pool for logging
         worker_pools = [
           # {
```
Comment (ocofaigh) [inline patterns/roks/module/config.tf:93]: I know this code block is commented out, but I think we should still add the `boot_volume_crk_name` variable to it incase we ever decide to uncomment it and create a worker pool

**Review by ocofaigh — APPROVED** [review]

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**